akka-http-core/src/main/scala/akka/http/impl/engine/ws/WebSocket.scala

```diff
@@ -64,20 +65,24 @@ private[http] object WebSocket {
   def messageAPI(serverSide: Boolean,
                  closeTimeout: FiniteDuration): BidiFlow[FrameHandler.Output, Message, Message, FrameOutHandler.Input, NotUsed] = {
     /* Completes this branch of the flow if no more messages are expected and converts close codes into errors */
-    class PrepareForUserHandler extends PushStage[MessagePart, MessagePart] {
-      var inMessage = false
-      def onPush(elem: MessagePart, ctx: Context[MessagePart]): SyncDirective = elem match {
-        case PeerClosed(code, reason) ⇒
-          if (code.exists(Protocol.CloseCodes.isError)) ctx.fail(new PeerClosedConnectionException(code.get, reason))
-          else if (inMessage) ctx.fail(new ProtocolException(s"Truncated message, peer closed connection in the middle of message."))
-          else ctx.finish()
-        case ActivelyCloseWithCode(code, reason) ⇒
-          if (code.exists(Protocol.CloseCodes.isError)) ctx.fail(new ProtocolException(s"Closing connection with error code $code"))
-          else ctx.fail(new IllegalStateException("Regular close from FrameHandler is unexpected"))
-        case x: MessageDataPart ⇒
-          inMessage = !x.last
-          ctx.push(x)
-        case x ⇒ ctx.push(x)
+    class PrepareForUserHandler extends SimpleLinearGraphStage[MessagePart] {
```

Comment:
Does this need to be a class, couldn't it be defined as an `object` outside of this method?